src/peg/testpeg.py

```diff
@@ -48,7 +48,10 @@ def get_parser(debug=False, root='query_specification'):
     peg_rules = re.sub('#', '// ', peg_rules)
 
     # adapt character range syntax
-    peg_rules = re.sub("'\\[", "r'[", peg_rules)
+    peg_rules = re.sub("\\[", "r'[", peg_rules)
+    peg_rules = re.sub("\\!r'\\[", "r'[^", peg_rules)
+    peg_rules = re.sub("\\]", "]'", peg_rules)
+    peg_rules = re.sub("EOF <-[^;]*;", "", peg_rules)
```

Comment:
I'm not sure I'm too enthused by hacking the grammar syntax with regular expressions.  For now, that's probably all right, but we should have a plan to parse our grammar into a tree and then serialise it in the various syntaxes, I guess.  I just wonder if someone else might not already have written something like that...

Reply:
That's indeed a very interesting idea and much stable/reliable than regular expressions, though apparently this is enough and allowed us to quickly test this grammar with different parsers.